### PR TITLE
Add prettier + husky

### PR DESCRIPTION
**Pull request by mattrheault**
## Review Instructions: FYI

Give this PR description a read for background context, & leave a comment if you have opinions on shared styleguide + linting rules for this repo 😄 

## JS styleguides & prettier

Javascript & the web as a whole lack standards. While this has been rapidly improving in recent years, still to this day, the JS written at one company often looks wildly different than at another. This is why it's important to have a coding styleguide for JS apps _(at hubspot, we use the [airbnb styleguide](https://github.com/airbnb/javascript))_. Having a styleguide ensures that we're all writing the same "flavor" of javascript.

But manually enforcing a styleguide in PR reviews is tedious. This is where [prettier](https://prettier.io/) comes in! Prettier used w/ pre-commit hook tools like husky allows us to enforce our styleguide rules automatically without devs having to do anything. Simply write your code like you would normally, and prettier + husky will auto-transform your code to adhere to the styleguide upon commit.

**Note** - this change requires all devs contributing to ```/Client``` to pull from master & run ```npm install```.

Screen Shot 2020-08-06 at 10 44 07 AM

## JS linting & eslint

While styleguides & prettier help us enforce code syntax & formatting, linting tools help us avoid bugs, steer us away from using deprecated language mechanics, and generally help us enforce best practices. [Eslint](https://eslint.org/) is the preferred open source JS linter, and can be configured to run on commit with prettier.

Eslint is already installed in ```/Client```. But we'll want to customize the lint rules, add it to the husky pre-commit hook, & run eslint on the entire project to make sure our code passes lint validation. I'll do this in a separate follow-up PR since this ones already pretty big!

### CODE REVIEW

File: Client/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (mattrheault) [inline Client/.prettierrc:1]: This is just the default [basic configuration](https://prettier.io/docs/en/configuration.html#basic-configuration) for prettier. We can get as specific here as we want. At hubspot, we surveyed all of our frontend devs to arrive at a shared prettier config that we're all comfortable with.

File: Client/package.json [hunk 2]
```diff
@@ -12,9 +12,6 @@
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.48",
     "@material-ui/pickers": "^3.2.10",
-    "@testing-library/jest-dom": "^4.2.4",
```
Comment (mattrheault) [inline Client/package.json:15]: These deps were giving me ```npm install``` troubles, but aren't actually used. Removing for now. We can re-add jest + enzyme if / when we add proper unit tests for our react app.

**Comment by ishchhabra** [comment]
Does the prettier + husky also work on backend code because if not shouldn't we merge it to frontend-dev rather than the master?

**Comment by mattrheault** [comment]
@ishchhabra great point about merge conflicts. For what it's worth, there are strategies that can be used to auto-resolve conflicts on all of the open branches, but it'd be more effort than it's worth to go down that path here. I'm gonna close this PR (and the [eslint pr](https://github.com/cicscareers/320-S20-Track1/pull/427)) in favor of https://github.com/cicscareers/320-S20-Track1/pull/428 so that we can roll this out with minimal pain.

## Rollout plan

- [ ] **Step 1:** merge prettier + eslint config into master (https://github.com/cicscareers/320-S20-Track1/pull/428)
- [ ] **Step 2:** all devs pull from master + run ```npm install``` in ```Client/```

At this point in time, the commit hook will auto-format any staged files that we try to commit. For example, if I make a 1-line change to **appts.js**, and try to commit that change, then prettier + eslint will auto-format that module. This will prevent future formatting & lint errors, without retroactively reformatting the entire codebase.

- [ ] **Step 3:** once all complex FE branches that pre-date prettier + eslint have been merged into master, we will retroactively run prettier + eslint on all of ```/Client/src```.

**Pull request closed without merging**